### PR TITLE
Util: Modify SplitHostPortDefault not to return an error for empty input

### PR DESCRIPTION
**Pull request by papagian**
**What this PR does / why we need it**:
Due to a recent change the SQL Server tests failed because passing an empty datasource url in `util.SplitHostPortDefault` was no more allowed.
This fix contains the following modifications:
- Modifies the `util.SplitHostPortDefault` not to return an error for empty input.
- Modifies the `util.SplitHostPort` to return an error for empty input.
- Introduces an additional test.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/util/ip_address.go [hunk 1]
```diff
@@ -38,12 +38,12 @@ type NetworkAddress struct {
...
 
 // SplitHostPortDefault splits ip address/hostname string by host and port. Defaults used if no match found
-func SplitHostPortDefault(input, defaultHost, defaultPort string) (NetworkAddress, error) {
+func SplitHostPortDefault(input, defaultHost, defaultPort string, allowEmpty bool) (NetworkAddress, error) {
 	addr := NetworkAddress{
 		Host: defaultHost,
 		Port: defaultPort,
 	}
-	if len(input) == 0 {
+	if len(input) == 0 && !allowEmpty {
```
Comment (marefr) [inline pkg/util/ip_address.go]: To me the correct handling if `input` is empty would be to return `addr`. What do you think?

**Review by marefr — COMMENTED** [review]
Doesn't feel correct to provide one additional argument `allowEmpty` when the name of the function is "SplitHostPortDefault` and you provide defaults? Added a suggestion in comment.

**Comment by papagian** [comment]
> Doesn't feel correct to provide one additional argument `allowEmpty` when the name of the function is "SplitHostPortDefault` and you provide defaults? Added a suggestion in comment.

The expected behaviour for me would be not to check at all the input since there are defaults.

**Comment by marefr** [comment]
@papagian 

> The expected behaviour for me would be not to check at all the input since there are defaults.

guess I tried to describe that in https://github.com/grafana/grafana/pull/20351#discussion_r345726031

File: pkg/util/ip_address.go [hunk 2]
```diff
@@ -82,5 +82,8 @@ func SplitHostPortDefault(input, defaultHost, defaultPort string) (NetworkAddres
 
 // SplitHostPort splits ip address/hostname string by host and port
 func SplitHostPort(input string) (NetworkAddress, error) {
+	if len(input) == 0 {
```
Comment (papagian) [inline pkg/util/ip_address.go]: Otherwise [this](https://github.com/grafana/grafana/blob/master/pkg/util/ip_address_test.go#L38) test fails.

Comment (marefr) [reply pkg/util/ip_address.go]: Make sense :+1:

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**